src/lavinmq/clustering/server.cr

```diff
@@ -160,12 +160,16 @@ module LavinMQ
         end
       rescue ex : AuthenticationError
         Log.warn { "Follower negotiation error" }
+        socket.close
       rescue ex : InvalidStartHeaderError
         Log.warn { ex.message }
+        socket.close
       rescue ex : IO::EOFError
         Log.info { "Follower disconnected" }
+        socket.close
       rescue ex : IO::Error
         Log.warn { "Follower disonnected: #{ex.message}" }
+        socket.close
       ensure
         follower.try &.close
```

Comment:
shouldn't `Follower#close` close the socket?

Reply:
It will be done by the action loop once it's drained. But we should probably close the socket there is no lag or if we get a timeout waiting for the follower to sync. I guess we could add an ensure statement and always close the socket in `#close`.